### DISCUSSION TITLE
Serverless Functions Work on Prod but not Local

### DISCUSSION THREAD

**Discussion by yiyun-sj** [vercel/vercel#8266] [Help]
Running a super simple python serverless function under /api/index.py
![image](https://user-images.githubusercontent.com/67709970/181423881-2234be38-1fcb-4870-9e91-41739a02e600.png)

When I deployed the code to vercel it seems to work just fine, but if I do `vercel dev` with the CLI, I get
![image](https://user-images.githubusercontent.com/67709970/181424123-8bf2120c-6e97-488b-ac9e-59648a46153e.png)

and in my terminal logs:

`<class 'OSError'>`
`LambdaError: Unspecified runtime initialization error`
`    at Lambda.<anonymous> (C:\Users\samyi\AppData\Roaming\npm\node_modules\vercel\dist\index.js:9811:27)`
`    at Generator.next (<anonymous>)`
`    at fulfilled (C:\Users\samyi\AppData\Roaming\npm\node_modules\vercel\dist\index.js:9725:58)`
`    at runMicrotasks (<anonymous>)`
`    at processTicksAndRejections (node:internal/process/task_queues:96:5)`

Any ideas on what's happening? Thank you!

**Comment by amyegan** [comment]
Hi @yiyun-sj. Are you using Windows, or a different operating system? Another community member was able to solve a Python local dev OS error on Windows by installing [Windows Subsystem for Linux](https://docs.microsoft.com/en-us/windows/wsl/about). 

https://github.com/vercel/vercel/discussions/8140#discussioncomment-3130070

**Reply by yiyun-sj** [reply]
Hi @amyegan! Thanks for the quick response. I will give this a try!

**Reply by yiyun-sj** [reply]
Hi @amyegan thank you! Gave this a try and it worked!

**Reply by dmarcucci** [reply]
Hi @amyegan,

Unfortunately, it seems like WSL2 has its own problem that makes it unsuitable as a solution for me (and I'm sure others). When running from there, the app is not exposed to `localhost` by default. There are [workarounds](https://github.com/microsoft/WSL/issues/4150#issuecomment-504209723) for this, such as using `portproxy` to port forward from `localhost` to the WSL environment via the IP for its ethernet adapter (`eth0`). But because this is a virtual network adapter, the IP is subject to change (and does so quite often).

Fortunately, I believe I found a solution to the Windows issue which allows for running `vercel dev` natively without the aforementioned issue. However, I'm not sure if this fix would break other operating systems.

The error cited above seems to be triggered by [this line](https://github.com/vercel/vercel/blob/main/packages/python/vc_init.py#L61) in `vc_init.py` from the Python runtime. The reason for this seems to be due to the `HTTPConnection` instance from [the line before](https://github.com/vercel/vercel/blob/main/packages/python/vc_init.py#L60). The address `0.0.0.0` does not work and I believe is what causes the error. After changing this to `127.0.0.1`, everything worked for me as expected.

Is this something that the team at Vercel could look into? I'm sure this would help many people and having the option to bypass WSL for Windows users would be ideal.

Thanks.

**Comment by dmarcucci** [comment]
Looks like this was fixed in https://github.com/vercel/vercel/pull/8880 for anybody that was still running into this issue.

**Reply by amyegan** [reply]
Thanks for coming back to share this update!
